File: 백준/11657/solution.py
import sys
input = sys.stdin.readline
INF = sys.maxsize

N, M = map(int, input().split())

# 거리 저장
dist = [INF] * (N+1)
# 간선 정보
edges = []
for _ in range(M):
    a, b, w = map(int, input().split())
    edges.append((a, b, w))

def belmanFord(start):
    dist[start] = 0
    for i in range(N):
        for j in range(M):
            cur, nxt, weight = edges[j]
            if dist[cur] != INF and dist[nxt] > dist[cur] + weight:
                dist[nxt] = dist[cur] + weight
                if i == N-1:
                    return True
    return False


isCycle = belmanFord(1)
if isCycle:
    print("-1")
else:
    for i in range(2, N+1):
        if dist[i] == INF:
            print("-1")
        else:
            print(dist[i])

File: 백준/11657/solution.swift
typealias Data = (cur: Int, nxt: Int, weight: Int)
let NM = readLine()!.split(separator: " ").compactMap{Int($0)}
let N = NM[0]
let M = NM[1]
var edges: [Data] = []
for _ in 0..<M {
    let abc = readLine()!.split(separator: " ").compactMap{Int($0)}
    edges.append((abc[0], abc[1], abc[2]))
}
let INF = Int.max
var dist = [Int](repeating: INF, count: N+1)

func bellmanFord(_ src: Int) -> Bool {
    dist[src] = 0
    for i in 0..<N {
        for j in 0..<M {
            let data = edges[j]
            if dist[data.cur] != INF && dist[data.nxt] > dist[data.cur] + data.weight {
                dist[data.nxt] = dist[data.cur] + data.weight
                if i == N-1 {
                    return true
                }
            }
        }
    }
    return false
}
let isCycle = bellmanFord(1)
if isCycle {
    print("-1")
} else {
    for i in 2...N {
        if dist[i] == INF {
            print("-1")
        }
        else {
            print(dist[i])
        }
    }
}
    

